plugins/aggregators/basicstats/basicstats.go

```diff
@@ -222,45 +220,38 @@ func parseStats(names []string) *configuredStats {
 			parsed.non_negative_diff = true
 
 		default:
-			log.Printf("W! Unrecognized basic stat '%s', ignoring", name)
+			b.Log.Warnf("unrecognized basic stat '%s', ignoring", name)
 		}
 	}
 
 	return parsed
 }
 
-func defaultStats() *configuredStats {
-
-	defaults := &configuredStats{}
-
-	defaults.count = true
-	defaults.min = true
-	defaults.max = true
-	defaults.mean = true
-	defaults.variance = true
-	defaults.stdev = true
-	defaults.sum = false
-	defaults.non_negative_diff = false
-
-	return defaults
+var defaultStats = &configuredStats{
```

Comment:
I wouldn't do this, as it means all instances share the same configuredStats instance.  If any plugin is modified in the future to modify this struct it will affect all plugin instances.  So I feel this is playing with fire.

If you want to simplify this, I'd call getConfiguredStats() from the Init() function and have it return/set a non-pointer struct.  If we initialize it in Init we don't need to check it for nil on Push anymore.  The `parseStats(` is probably better as a non-member function.

Reply:
D'oh! good catch